### PR TITLE
Create a new module proxy-utils

### PR DESCRIPTION
**Pull request by rafabene**

### CODE REVIEW

File: deltaspike/modules/proxy-utils/api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Licensed to the Apache Software Foundation (ASF) under one or more contributor
```
Comment (johnament) [inline deltaspike/modules/proxy-utils/api/pom.xml]: license header looks a bit off.

File: deltaspike/modules/proxy-utils/api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>2.3</version>
+                <configuration>
+                    <shadedArtifactAttached>false</shadedArtifactAttached>
+                    <createDependencyReducedPom>false</createDependencyReducedPom>
+                    <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
+                    <relocations>
+                        <relocation>
+                            <pattern>org.objectweb.asm</pattern>
+                            <shadedPattern>org.apache.deltaspike.core.util.proxy.asm5</shadedPattern>
```
Comment (johnament) [inline deltaspike/modules/proxy-utils/api/pom.xml]: Can we continue to use the proxyutil package?

File: deltaspike/modules/proxy-utils/api/pom.xml (hunk 1 above)
Comment (rafabene) [inline deltaspike/modules/proxy-utils/api/pom.xml]: Thanks... Will fix it

File: deltaspike/modules/proxy-utils/api/pom.xml (hunk 2 above)
Comment (rafabene) [inline deltaspike/modules/proxy-utils/api/pom.xml]: I'm not sure. Anyway, I think I should change this package.

**Pull request merged**